### PR TITLE
feat(vital-carousel): Add vital carousel (JEPZ-476)

### PR DESCRIPTION
**Pull request by hvanyo**

### CODE REVIEW

File: packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+      addProps({
+        isPlaying: true,
+      }),
+    ),
+  }
+});
+
+// TODO (now to pass in tokens:  (interval: number = 3000) )
+const WithAutoPlayInterval = asVitalCarouselToken({
+// const withAutoPlayInterval = (interval: number = 3000) => withDesign({
```
Comment (hvanyo) [inline packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:51]: @marcopagliarulo @wodenx   the original carousel token pass allowed a prop.  Do we have example of correct way to this with tokens?  There is another example below that passes in nodekeys at line 89

Comment (marcopagliarulo) [reply packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:51]: I don't think we have any "dynamic token", btw I don't see any cons to use something like that
`
const WithAutoPlayInterval = (interval) => asVitalCarouselToken({
  Behavior: {
    Wrapper: addProps({
      interval,
    }),
  },
})
`

Comment (hvanyo) [reply packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:51]: @marcopagliarulo 
`
@bodiless/vital-carousel: src/components/Carousel/tokens/vitalCarousel.ts(252,3): error TS2322: Type '(nodeKeys?: WithNodeKeyProps) => TokenSpec<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>' is not assignable to type 'VitalCarouselToken'.
@bodiless/vital-carousel:   Type '(nodeKeys?: WithNodeKeyProps) => TokenSpec<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>' is not assignable to type 'Required<FinalDomains<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>>'.
@bodiless/vital-carousel: src/components/Carousel/tokens/vitalCarousel.ts(253,3): error TS2322: Type '(interval?: number) => TokenSpec<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>' is not assignable to type 'VitalCarouselToken'.
@bodiless/vital-carousel:   Type '(interval?: number) => TokenSpec<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>' is not assignable to type 'Required<FinalDomains<VitalCarouselComponents, { Core: {}; Components: {}; A11y: {}; Analytics: {}; SEO: {}; Layout: {}; Spacing: {}; Theme: {}; A11yContent: {}; Content: {}; Behavior: {}; Schema: {}; }>>'.
`

Comment (hvanyo) [reply packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:51]: actually I think in my use-cases I don't need the ability to change these props..

File: packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  }
+});
+
+// const WithCarouselDots = (nodeKeys?: WithNodeKeyProps) => asVitalCarouselToken({
+const WithControls = asVitalCarouselToken({
+  Components: {
+    Dots: as(
+      replaceWith(Ul),
+      asBodilessList(CAROUSEL_NODE_KEY, undefined, () => ({ groupLabel: 'Slide' })),
+      withDesign({
```
Comment (hvanyo) [inline packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:89]: @wodenx  all the controls do a lot of driving in list for styling and all done with design..  I am not sure there is value to separate all these out so we don't use withdesign?  Your opinion. Applies to this token and next two.

**Pull request closed without merging**